hal/src/stm32f2xx/sleep_hal.cpp

```diff
@@ -306,6 +445,44 @@ static int enterStopBasedSleep(const hal_sleep_config_t* config, hal_wakeup_sour
         } else if (wakeupSource->type == HAL_WAKEUP_SOURCE_TYPE_GPIO) {
             auto gpioWakeup = reinterpret_cast<hal_wakeup_source_gpio_t*>(wakeupSource);
             HAL_Interrupts_Detach_Ext(gpioWakeup->pin, 1, nullptr);
+        } else if (wakeupSource->type == HAL_WAKEUP_SOURCE_TYPE_USART) {
+            auto usartWakeup = reinterpret_cast<hal_wakeup_source_usart_t*>(wakeupSource);
+            // Enabled unwanted interrupts and unbump the interrupt priority
+            NVIC_InitTypeDef NVIC_InitStructure;
+            switch (usartWakeup->serial) {
+                case HAL_USART_SERIAL1: {
+                    NVIC_InitStructure.NVIC_IRQChannel = USART1_IRQn;
+                    USART_ITConfig(USART1, USART_IT_TXE, ENABLE); // TXE pending bit is cleared only by a write to the USART_DR register
+                    break;
+                }
+                case HAL_USART_SERIAL2: {
+                    NVIC_InitStructure.NVIC_IRQChannel = USART2_IRQn;
+                    USART_ITConfig(USART2, USART_IT_TXE, ENABLE);
+                    break;
+                }
+#if PLATFORM_ID == PLATFORM_ELECTRON_PRODUCTION
+                case HAL_USART_SERIAL3: {
+                    NVIC_InitStructure.NVIC_IRQChannel = USART3_IRQn;
+                    USART_ITConfig(USART3, USART_IT_TXE, ENABLE);
+                    break;
+                }
+                case HAL_USART_SERIAL4: {
+                    NVIC_InitStructure.NVIC_IRQChannel = UART4_IRQn;
+                    USART_ITConfig(UART4, USART_IT_TXE, ENABLE);
+                    break;
+                }
+                case HAL_USART_SERIAL5: {
+                    NVIC_InitStructure.NVIC_IRQChannel = UART5_IRQn;
+                    USART_ITConfig(UART5, USART_IT_TXE, ENABLE);
+                    break;
+                }
+#endif
+                default: return SYSTEM_ERROR_INVALID_ARGUMENT; // It should not reach here.
+            }
+            NVIC_InitStructure.NVIC_IRQChannelPreemptionPriority = 7;
```

Comment:
Nit: magic number.

Reply:
Ideally, this value should be the same as what we set in `usart_hal.c`, But there isn't a variable or a macro for this value in usart_hal.c. So I'll just leave a comment here for better understanding.